fendermint/vm/topdown/src/sync.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! A constant running process that fetch or listener to parent state
+
+use crate::error::Error;
+use crate::{BlockHeight, Config, ParentFinalityProvider};
+use anyhow::{anyhow, Context};
+use ipc_sdk::cross::CrossMsg;
+use ipc_sdk::ValidatorSet;
+use std::cmp::max;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+
+/// Constantly syncing with parent through polling
+pub struct PollingParentSyncer<T> {
+    config: Config,
+    started: Arc<AtomicBool>,
+    parent_view_provider: Arc<T>,
+    agent: Arc<IPCAgentProxy>,
+}
+
+impl<T> PollingParentSyncer<T> {
+    pub fn new(config: Config, parent_view_provider: Arc<T>, agent: Arc<IPCAgentProxy>) -> Self {
+        Self {
+            config,
+            started: Arc::new(AtomicBool::new(false)),
+            parent_view_provider,
+            agent,
+        }
+    }
+}
+
+impl<T: ParentFinalityProvider + Send + Sync + 'static> PollingParentSyncer<T> {
+    /// Start the proof of finality listener in the background
+    pub fn start(&mut self) -> anyhow::Result<()> {
+        match self
+            .started
+            .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
+        {
+            Ok(_) => {}
+            Err(_) => return Err(anyhow!("already started")),
+        }
+
+        let config = self.config.clone();
+        let provider = self.parent_view_provider.clone();
+        let agent = self.agent.clone();
+
+        tokio::spawn(async move {
+            loop {
+                if let Err(e) = sync_with_parent(&config, &agent, &provider).await {
+                    tracing::info!("sync with parent encountered error: {e}");
+                }
+
+                // TODO: add termination channel?
```

Comment:
I think it is useful to expose some way to stop the process and make it up to the user to use it or not.

Reply:
I currently removed this `todo` as I'm not so sure how/when should be stop this process. I assume this should run as long as `fendermint` is up? I will add the necessary code when it starts to get clearer.